packages/grid/src/vaadin-grid-selection-column-base-mixin.js

```diff
@@ -201,10 +206,14 @@ export const GridSelectionColumnBaseMixin = (superClass) =>
 
     /** @private */
     __onGridInteraction(e) {
-      this._shiftKeyDown = e.shiftKey;
+      if (e instanceof KeyboardEvent) {
+        this._shiftKeyDown = e.key !== 'Shift' && e.shiftKey;
```

Comment:
**note:** I added a connector test that makes sure `$server.setShiftKeyDown` isn't called when just pressing <kbd>Shift</kbd> without selecting anything.